tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -226,16 +232,277 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
       emitNextLine(0)
   }
 
+  // Private API for Rhino
+
+  private[scalajs] object rhinoAPI { // scalastyle:ignore
+    private implicit val GlobalKnowledge: GlobalKnowledge = new Invalidatable {
```

Comment:
It's probably clearer with `KnowledgeAccessor`, indeed.

It really isn't about creating other forwarders in `KnowledgeGuardian`. This should be seen as a sin, since we would have a guardian that is not actually guarding anything, since it offers an API that doesn't actually track anything. It is the logical thing to do in `rhinoAPI` to fake the tracking, because we know that Rhino doesn't reuse an `Emitter` twice. I should write that as a comment, in fact ^^